src/main/java/decodex/modules/ModuleManager.java

```diff
@@ -1,51 +1,189 @@
+            module = prepareHexDecoder();
+            break;
+        case BinaryEncoder.MODULE_NAME:
+            hasNoParameters(parameters);
+            module = prepareBinaryEncoder();
+            break;
+        case BinaryDecoder.MODULE_NAME:
+            hasNoParameters(parameters);
+            module = prepareBinaryDecoder();
+            break;
+        case RotEncoder.MODULE_NAME:
+            module = prepareRotEncoder(parameters);
+            break;
+        default:
+            throw new UnknownModuleException(moduleName);
+        }
+
+        logger.fine(String.format("Selected %s", module.getName()));
+        return module;
+    }
+
+    /**
+     * Throws ModuleException if the module parameters are empty.
+     *
+     * @param parameters Parameters for specific commands.
+     * @throws ModuleException If the String array of module parameters is not empty.
+     */
+    private void hasNoParameters(String[] parameters) throws ModuleException {
+        if (parameters.length > 0) {
+            throw new ModuleException(ErrorMessages.TOO_MANY_MODULE_PARAMETERS);
+        }
+    }
+
+    /**
+     * Prepares and returns the Base64Encoder.
+     *
+     * @return The Base64Encoder object.
+     */
+    private Module prepareBase64Encoder() {
+        return new Base64Encoder();
+    }
+
+    /**
+     * Prepares and returns the Base64Decoder.
+     *
+     * @return The Base64Decoder object.
+     */
+    private Module prepareBase64Decoder() {
+        return new Base64Decoder();
+    }
+
+    /**
+     * Prepares and returns the HexEncoder.
+     *
+     * @return The HexEncoder object.
+     */
+    private Module prepareHexEncoder() {
+        return new HexEncoder();
+    }
+
+    /**
+     * Prepares and returns the HexDecoder.
+     *
+     * @return The HexDecoder object.
+     */
+    private Module prepareHexDecoder() {
+        return new HexDecoder();
+    }
+
+    /**
+     * Prepares and returns the BinaryEncoder.
+     *
+     * @return The BinaryEncoder object.
+     */
+    private Module prepareBinaryEncoder() {
+        return new BinaryEncoder();
+    }
+
+    /**
+     * Prepares and returns the BinaryDecoder.
+     *
+     * @return The BinaryDecoder object.
+     */
+    private Module prepareBinaryDecoder() {
+        return new BinaryDecoder();
+    }
+
+    /**
+     * Prepares and returns the RotEncoder with specified rotation.
+     *
+     * @param parameters String array of module parameters.
+     * @return The RotEncoder object.
+     * @throws ModuleException If the number of parameters are incorrect or invalid.
+     */
+    private Module prepareRotEncoder(String[] parameters) throws ModuleException {
+        if (parameters.length == 0) {
+            throw new ModuleException(ErrorMessages.MISSING_MODULE_PARAMETERS);
+        } else if (parameters.length > 1) {
+            throw new ModuleException(ErrorMessages.TOO_MANY_MODULE_PARAMETERS);
         }
```

Comment:
Perhaps remove the `else` keyword and move this out as a standalone `if` clause?